drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -99,15 +166,21 @@ static enum clock_control_status clock_control_nrf_auxpll_get_status(const struc
 	return CLOCK_CONTROL_STATUS_OFF;
 }
 
-static DEVICE_API(clock_control, clock_control_nrf_auxpll_api) = {
-	.on = clock_control_nrf_auxpll_on,
-	.off = clock_control_nrf_auxpll_off,
-	.get_rate = clock_control_nrf_auxpll_get_rate,
-	.get_status = clock_control_nrf_auxpll_get_status,
-};
-
 static int clock_control_nrf_auxpll_init(const struct device *dev)
 {
+	struct dev_data_auxpll *dev_data = dev->data;
+	static const struct onoff_transitions transitions = {
+		.start = onoff_start_auxpll,
+		.stop = onoff_stop_auxpll
+	};
```

Comment:
please move static data outside of function, its placed in ROM anyway so it does not affect the stack, and the scope of the file is limited